docs/CODE_REVIEW.md

```diff
@@ -0,0 +1,145 @@
+# How to do code reviews for curl
+
+Anyone and everyone is encouraged and welcome to review code submissions in
+curl. This is a guide on what to check for and how to perform a successful
+code review.
```

Comment:
I suggest that we add a blurb about how we really want anyone who is interested to join, not just C experts. I'm not sure how to word it, but perhaps something like "Code review is not exclusively for subject matter experts or curl maintainers, everyone at every level is welcome. If a PR isn't understandable then asking for clarification, or more code comments, is a perfectly valid review input."

Reply:
Isn't that just saying the same thing as *Anyone and everyone is encouraged and welcome to review code submissions in curl.* but with more words? Do we really need to spell it out like that?

Reply:
Perhaps we don't no.